mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -524,6 +527,85 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
         }
     }
 
+    private fun selectServerBasedOnWifi() {
+        fun getPermissionForSsidRead() = when {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q -> Manifest.permission.ACCESS_FINE_LOCATION
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.O -> Manifest.permission.ACCESS_COARSE_LOCATION
+            else -> null
+        }
+
+        val anyServerHasSetWifi = prefs
+            .getConfiguredServerIds()
+            .map { id -> ServerConfiguration.load(prefs, getSecretPrefs(), id) }
+            .any { config -> config?.wifiSsid?.isNotEmpty() == true }
+
+        val locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
+        val wifiManager = applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
+        val wifiInfo = wifiManager.connectionInfo
+        val wifiSsid = if (wifiInfo.networkId == -1) null else wifiInfo.ssid.removeSurrounding("\"")
```

Comment:
Can't you shorten this to `val wifiSsid = wifiManager.connectionInfo.ssid?.removeSurrounding("\"")`?

Reply:
`getSSID()` always returns a non-null string:

> The SSID may be {@link WifiManager#UNKNOWN_SSID}, if there is no network currently connected or if the caller has insufficient permissions to access the SSID.

````java
public static final String UNKNOWN_SSID = "<unknown ssid>";
````